### DISCUSSION TITLE
Método readLineSync dúvida

### DISCUSSION THREAD

**Discussion by marcos1martins** [Flutterando/forum#142] [Perguntas e Respostas]
por que quando rodo esse código, ele não passa da linha que lê o teclado?
```
import 'dart:io';
import 'dart:math' as math;

main() {
  print("Informe o raio:");
  double raio = stdin.readLineSync() as double;
  double circunferencia = 2 * math.pi * raio;
  print("A circunferencia tem " + circunferencia.toString());
}
```
li na documentação que é isso:
Blocks until a full line is available.

Lines my be terminated by either <CR><LF> or <LF>. On Windows, in cases where the [stdioType] of stdin is [StdioType.terminal], the terminator may also be a single <CR>.

mas não sei o que fazer pra terminar a linha, não entendi esse` <CR>`. não era pra ser só enter?

**Comment by codetrindade** [comment]
Por padrão, os aplicativos são executados no **Console de Depuração**, "debugConsole" onde a entrada é usada como um REPL para avaliar expressões. Se você quiser interagir com o **stdin**, precisará configurar o aplicativo para ser executado no **Terminal**.


Você pode fazer isso com uma configuração de inicialização:
```
  "console": "terminal",
```

Ou você pode definir a configuração dart.cliConsole no VS Code:

```
 "dart.cliConsole": "terminal"
```
